platforms/cpu/src/CpuKernels.cpp

```diff
@@ -166,6 +166,12 @@ class CpuCalcForcesAndEnergyKernel::InitForceTask : public ThreadPool::Task {
                 posq[4*i+1] = (float) posData[i][1];
                 posq[4*i+2] = (float) posData[i][2];
             }
+        
+        // Check for invalid positions.
+        
+        for (int i = 4*start; i < 4*end; i++)
+            if (posq[i] != posq[i])
```

Comment:
Are you sure this won't get optimized away by the compiler?  And what about `inf`?  Does that not cause a segfault?

(I'm just curious here)